models/stdp_symmetric_connection.h

```diff
@@ -0,0 +1,300 @@
+   * \param e The event to send
+   * \param t_lastspike Point in time of last spike sent.
+   * \param cp common properties of all synapses (empty).
+   */
+  void send( Event& e, thread t, double_t t_lastspike, const CommonSynapseProperties& cp );
+
+
+  class ConnTestDummyNode : public ConnTestDummyNodeBase
+  {
+  public:
+    // Ensure proper overriding of overloaded virtual functions.
+    // Return values from functions are ignored.
+    using ConnTestDummyNodeBase::handles_test_event;
+    port
+    handles_test_event( SpikeEvent&, rport )
+    {
+      return invalid_port_;
+    }
+  };
+
+  void
+  check_connection( Node& s,
+    Node& t,
+    rport receptor_type,
+    double_t t_lastspike,
+    const CommonPropertiesType& )
+  {
+    ConnTestDummyNode dummy_target;
+
+    ConnectionBase::check_connection_( dummy_target, s, t, receptor_type );
+
+    t.register_stdp_connection( t_lastspike - get_delay() );
+  }
+
+  void
+  set_weight( double_t w )
+  {
+    weight_ = w;
+  }
+
+private:
+  double_t
+  facilitate_( double_t w, double_t kplus )
+  {
+    double_t norm_w = ( w / Wmax_ ) + ( lambda_ * eta_ * kplus );
+    return norm_w < 1.0 ? norm_w * Wmax_ : Wmax_;
+  }
+
+  double_t
+  depress_( double_t w )
+  {
+    double_t norm_w = ( w / Wmax_ ) - ( 2. * kappa_ * tau_ * lambda_ * eta_ );
+    return norm_w > 0.0 ? norm_w * Wmax_ : 0.0;
+  }
+
+  // data members of each connection
+  double_t weight_;
+  double_t tau_;
+  double_t lambda_;
+  double_t kappa_;
+  double_t eta_;
+  double_t Wmax_;
+  double_t Kplus_;
+};
+
+
+/**
+ * Send an event to the receiver of this connection.
+ * \param e The event to send
+ * \param t The thread on which this connection is stored.
+ * \param t_lastspike Time point of last spike emitted
+ * \param cp Common properties object, containing the stdp parameters.
+ */
+template < typename targetidentifierT >
+inline void
+STDPSymmetricConnection< targetidentifierT >::send( Event& e,
+  thread t,
+  double_t t_lastspike,
+  const CommonSynapseProperties& )
+{
+  // synapse STDP depressing/facilitation dynamics
+  //   if(t_lastspike >0) {std::cout << "last spike " << t_lastspike << std::endl ;}
+  double_t t_spike = e.get_stamp().get_ms();
+  // t_lastspike_ = 0 initially
+
+  // use accessor functions (inherited from Connection< >) to obtain delay and target
+  Node* target = get_target( t );
+  double_t dendritic_delay = get_delay();
+
+  // get spike history in relevant range (t1, t2] from post-synaptic neuron
+  std::deque< histentry >::iterator start;
+  std::deque< histentry >::iterator finish;
+  target->get_history( t_lastspike - dendritic_delay, t_spike - dendritic_delay, &start, &finish );
+
+  // presynaptic neuron j, post synaptic neuron i
+  // Facilitation for each post synaptic spike
+  // Wij = Wij + eta*xj
+  double_t minus_dt;
+  while ( start != finish )
+  {
+    minus_dt = t_lastspike - ( start->t_ + dendritic_delay );
+    ++start;
+    if ( minus_dt == 0 )
```

Comment:
Well, in the classical STDP, as you point out, delta_w is undefined when delta_t is zero since in the neighbourhood of zero, the values are vastly different.

In the case of this symmetric rule, however, this isn't the case - the value of delta_w in the neighbourhood of zero on either side is the same - in this version of stdp, there should be a maximum facilitation when delta_t is zero. (Figure 1C in the paper).

So, is it OK if I remove the (minus_dt == 0) conditional?